java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -44,7 +44,7 @@
   // Package exposed for sharing between AllocatorManger and BaseAllocator objects
   final String name;
   final RootAllocator root;
-  private final Object DEBUG_LOCK = DEBUG ? new Object() : null;
+  private final Object debugLock = DEBUG ? new Object() : null;
```

Comment:
I'm not sure if we want to be so strict with variable naming rules
`(naming) MemberName: Member name 'DEBUG_LOCK' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'.`

A lot of what I'm seeing looks fine to me, such as "nBytes", which gets the same error since it requires 2 lowercase letters in the front.  What are your thoughts @jacques-n @icexelloss ?

Reply:
Seems like this is basically a constant, which should be able to have different naming rules

Reply:
Yeah, there is a check for constants, statics and finals. It just isn't in our conf http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.html

A suggested pattern is "^log(ger)?|[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"

Reply:
That doesn't look like it would be too disruptive to add, I see a few `final int isSet = 1;` that would need to be changed, but most are already in caps

Reply:
I think we can allow both
```
final int isSet = 1
```

and 

```
final int MY_CONSTANT = 1
```

I think both are common and reasonable

Reply:
Sounds fine to me, I'll try to change to pattern to allow both